### PR TITLE
remove implicit fallthroughs (#1194)

### LINKED ISSUE #1194: Build fails at 70% reliably using emsdk
I hope this is the right place for this issue. After a great deal of trouble getting `sdk-incoming-64bit` to install successfully (running out of memory, etc.) I am now having trouble getting `binaryen-master-64bit` to install. Running `./emsdk install binaryen-master-64bit` fails consistently at the same point and I am unable to parse the error to find useful information.

I have uploaded the entire output of the above command to pastebin, which can be found here: https://pastebin.com/yf8Sewew .

In case it is useful, the output of running `emcc -v` is

```
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 1.37.21
clang version 4.0.0 (https://github.com/kripken/emscripten-fastcomp-clang.git 974b55fd84ca447c4297fc3b00cefb6394571d18) (https://github.com/kripken/emscripten-fastcomp.git 087c6b7b18b7b769d4ad8f2ac3e0dd0ae6b924c2) (emscripten 1.37.21 : 1.37.21)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/jaday/dev/wasm/emsdk/clang/fastcomp/build_incoming_64/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-pc-linux-gnu/7.2.0
Found candidate GCC installation: /usr/lib64/gcc/x86_64-pc-linux-gnu/7.2.0
Selected GCC installation: /usr/lib64/gcc/x86_64-pc-linux-gnu/7.2.0
Candidate multilib: .;@m64
Selected multilib: .;@m64
INFO:root:(Emscripten: Running sanity checks)
```

I am at a bit of a loss as to what to do next or how to approach debugging this. Have other people had similar problems? Is there something I can do to get a more verbose error message to try and figure out where the build fails?

I am new to emscripten and WebAssembly in general, so my apologies if my ignorance is obvious in this issue.

### PR DESCRIPTION
**Pull request by z2oh**
Fixes two implicit fall through warnings which caused building to fail.

Closes #1194

### CODE REVIEW

File: src/tools/translate-to-fuzz.h [hunk 1]
```diff
@@ -639,8 +639,8 @@ class TranslateToFuzzReader {
           }
         }
         switch (conditions) {
-          case 0: if (!oneIn(4)) continue;
-          case 1: if (!oneIn(2)) continue;
+          case 0: if (!oneIn(4)) continue; break;
```
Comment (dschuff) [inline src/tools/translate-to-fuzz.h]: I think having it all on one line like this makes this difficult to read. Could we do
```
case 0:
 if (!oneIn(4)) continue;
 break;
```
instead?

Comment (z2oh) [reply src/tools/translate-to-fuzz.h]: Good point. Changed in b377dde.

File: src/tools/translate-to-fuzz.h [hunk 2]
```diff
@@ -639,9 +639,14 @@ class TranslateToFuzzReader {
           }
         }
         switch (conditions) {
-          case 0: if (!oneIn(4)) continue;
-          case 1: if (!oneIn(2)) continue;
-          default: if (oneIn(conditions + 1)) continue;
+          case 0:
```
Comment (kripken) [inline src/tools/translate-to-fuzz.h]: since the body isn't on the same line, please add `{,}` for each (see other switches for the convention)

Comment (z2oh) [reply src/tools/translate-to-fuzz.h]: Fixed in 90364e3.

**Comment by kripken** [comment]
Perfect, thanks! Just waiting for CI to finish before merging.

**Comment by dschuff** [comment]
lol @z2oh sorry you hit a perfect storm of conflicting reviewers. This isn't the first time @kripken and I have clashed on exactly this bit of style :D

**Comment by kripken** [comment]
Oh sorry, I didn't notice there were previous comments - they're hidden in the diff view.

@dschuff, did we ever come to a consensus on that? Looking in the code most have the `{,}` but I don't remember details of discussion on the topic.

**Comment by dschuff** [comment]
I think you won mostly just by virtue of having written a lot of code in that style before I did, so I didn't want to change it all :)

**Comment by kripken** [comment]
Heh, ok :) Anyhow, I'm not super-invested in this - if most people prefer another convention let's switch (separate from this PR of course).

But just for the record (since I'm not sure where else this is documented), my logic is that this is similar to the case of
```
if (..) {
  ..
} else {
  ..
}
```
Without `{,}` there is a risk of getting things wrong, and switch cases also denote separate code paths like if arms.

**Comment by dschuff** [comment]
Oh hey, the translate-to-fuzz test is failing becuase... yeah this actually does change the behavior. Before, we had multiple rolls of the RNG to `continue` and now it's just one. Since that was probably the intended behavior, should we just auto-update the tests?

**Comment by kripken** [comment]
Yeah, I agree. Let's just run `./auto_update_tests.py` and add that change here.

**Comment by z2oh** [comment]
I ran `./auto_update_tests.py` and a subsequent run of `./check.py` passed on my system. Here is the output of `git diff`: https://hastebin.com/zayuraniho .

I know very little about the testing process of this project, so I can't really say if this is correct or not. Should I commit this change?

**Comment by dschuff** [comment]
That looks about right to me. It's just the one test that's affected, and it will be affected in a random way, so it's probably correct. I'd say go ahead and commit it, and if @kripken agrees with my assessment we'll merge.

**Comment by kripken** [comment]
Yeah, perfect. Thanks again!

**Pull request merged**